### PR TITLE
Scale and transform plane according to map file

### PR DESCRIPTION
**Pull request by iman512003**
This is to address #39.

### CODE REVIEW

File: smarts/core/smarts.py [hunk 1]
```diff
@@ -340,10 +340,32 @@ def _setup_bullet_client(self, client: bc.BulletClient):
         client.setGravity(0, 0, -9.8)
 
         plane_path = self._scenario.plane_filepath
+
+        bounding_box_lenght = (
```
Comment (liamchzh) [inline smarts/core/smarts.py:344]: Maybe move the size and center calculation to scenario.py?

**Comment by liamchzh** [comment]
Could you take a look at `test_controller_lane.py::test_lane_following_controller`, it failed.

File: smarts/core/scenario.py [hunk 2]
```diff
@@ -699,3 +699,11 @@ def _validate_assets_exist(self):
     @property
     def scenario_hash(self):
         return self._scenario_hash
+
+    @property
+    def bounding_box(self):
+        # <2D-BOUNDING_BOX>: four floats separated by ',' (<FLOAT>,<FLOAT>,<FLOAT>,<FLOAT>),
```
Comment (JVillella) [inline smarts/core/scenario.py:705]: Can say this more simply,
`# 2D bbox in format (xmin, ymin, xmax, ymax)`

File: smarts/core/smarts.py (hunk 1 above)
Comment (JVillella) [reply smarts/core/smarts.py:344]: spelling `bounding_box_lenght` --> `bounding_box_length`

File: smarts/core/smarts.py [hunk 3]
```diff
@@ -340,10 +340,32 @@ def _setup_bullet_client(self, client: bc.BulletClient):
...
                 plane_path = str(path.absolute())
-        self._ground_bullet_id = client.loadURDF(plane_path, useFixedBase=True)
+
+        # The plane_scale is multiplied by 1.2 to ensure that plane
+        # is completely covering the map.
+        self._ground_bullet_id = client.loadURDF(
+            plane_path,
+            useFixedBase=True,
+            basePosition=bounding_box_center,
+            globalScaling=1.2 * plane_scale,
```
Comment (JVillella) [inline smarts/core/smarts.py:367]: Seems fine; adding a fixed padding makes more sense to me than multiplying.

File: smarts/core/smarts.py [hunk 4]
```diff
@@ -340,10 +340,32 @@ def _setup_bullet_client(self, client: bc.BulletClient):
...
+            self._scenario.bounding_box[2] - self._scenario.bounding_box[0]
+        )
+        bounding_box_width = (
+            self._scenario.bounding_box[3] - self._scenario.bounding_box[1]
+        )
+        bounding_box_center = [
+            (self._scenario.bounding_box[0] + self._scenario.bounding_box[2]) / 2,
+            (self._scenario.bounding_box[1] + self._scenario.bounding_box[3]) / 2,
+            0,
+        ]
```
Comment (JVillella) [inline smarts/core/smarts.py:354]: Can you move this up to the Scenario.bounding_box() method?

The return can be `bbox, width, height`, or `bbox_origin, width, height`.

File: smarts/core/smarts.py [hunk 5]
```diff
@@ -348,10 +348,21 @@ def _setup_bullet_client(self, client: bc.BulletClient):
...
         if not os.path.exists(plane_path):
             with pkg_resources.path(models, "plane.urdf") as path:
                 plane_path = str(path.absolute())
-        self._ground_bullet_id = client.loadURDF(plane_path, useFixedBase=True)
+
+        self._ground_bullet_id = client.loadURDF(
+            plane_path,
+            useFixedBase=True,
+            basePosition=self._scenario.bounding_box[2],
+            globalScaling=plane_scale,
```
Comment (liamchzh) [inline smarts/core/smarts.py:364]: Make it a bit larger than the map? Either padding or scaling.

File: smarts/core/scenario.py [hunk 6]
```diff
@@ -697,3 +697,20 @@ def _validate_assets_exist(self):
     @property
     def scenario_hash(self):
         return self._scenario_hash
+
+    @property
+    def bounding_box(self):
+        # This function returns the following tuple:
+        # (bbox length, bbox width,bbox center)
```
Comment (liamchzh) [inline smarts/core/scenario.py:704]: Missing a space between `bbox width` and `bbox center`.

**Review by liamchzh — CHANGES_REQUESTED** [review]

File: smarts/core/scenario.py [hunk 7]
```diff
@@ -697,3 +697,20 @@ def _validate_assets_exist(self):
     @property
     def scenario_hash(self):
         return self._scenario_hash
+
+    @property
+    def bounding_box(self):
```
Comment (JVillella) [inline smarts/core/scenario.py:702]: map_bounding_box?

**Review by JVillella — APPROVED** [review]
Minor comments, but it's good to merge.

**Review by liamchzh — APPROVED** [review]

**Pull request merged**